### DISCUSSION TITLE
How to add dcim interface

### DISCUSSION THREAD

**Discussion by yuzhaninan** [benclaussen/NetboxPS#32] [General]
Add-NetboxDCIMInterface does not have a Type parameter that is required for creation. You have to use the construct

$TempBody = @{
                                                                name = 'Network Adapter 1'
                                                                type = 'other'
                                                                device = $OTPNetboxSystemRecord.ID
                                                        }
                                                        $OTPNetboxSystemRecordInterface = InvokeNetboxRequest -URI "https://$(Get-NetboxHostname)/api/dcim/interfaces/" -Body $TempBody -method:post

**Comment by alagoutte** [comment]
What netbox release do you are using ?

may be a new parameter of last release...

**Reply by yuzhaninan** [reply]
Get-NetboxVersion


django-version     : 4.0.4
installed-apps     : @{debug_toolbar=3.2.4; django_filters=21.1; django_prometheus=2.2.0; django_rq=2.5.1; django_tables2=2.4.1; drf_yasg=1.20.0; graphene_django=2.
                     15.0; graphiql_debug_toolbar=0.2.0; mptt=0.13.4; rest_framework=3.13.1; social_django=5.0.0; taggit=2.1.0; timezone_field=5.0}
netbox-version     : 3.2.2
plugins            :
python-version     : 3.9.2
rq-workers-running : 1

**Comment by alagoutte** [comment]
Yes missing type field on https://github.com/benclaussen/NetboxPS/blob/dev/Functions/DCIM/Interfaces/Add-NetboxDCIMInterface.ps1#L18

Can you push a PR for add the field ?

**Reply by FluffyBeaver** [reply]
And it seems that now (I have Nebox 3.1.5) the parameter 'mode' could accept only this vaue: [ access, tagged, tagged-all ].
So code on https://github.com/benclaussen/NetboxPS/blob/dev/Functions/DCIM/Interfaces/Add-NetboxDCIMInterface.ps1#L54 should be deleted. 
But I don't know how it worked on older versions...

**Reply by alagoutte** [reply]
Ok, i will check...

**Comment by yuzhaninan** [comment]
It will take some time

**Reply by alagoutte** [reply]
i can push if you want !

**Reply by yuzhaninan** [reply]
I don't know how to post such a big fix.
[ArgumentCompleter({(Get-NetboxAPIDefinition).definitions.WritableInterface.properties.type.enum})][string]$Type = 'other'

?

**Reply by alagoutte** [reply]
I will look

(not sure to add check value for the start ;)
